genesis-builder/framework/erc20/sources/erc20.move

```diff
@@ -95,5 +105,142 @@ module Erc20::erc20 {
         approve(caller, token, spender, value);
     }
 
-    // TODO: transfer, transferFrom, etc.
+    struct TransferArgs {
+        recipient: address,
+        value: u256,
+    }
+
+    public fun transfer(
+        caller: &signer,
+        token: address,
+        recipient: address,
+        value: u256,
+    ): EvmResult {
+        let args = TransferArgs {
+            recipient,
+            value,
+        };
+
+        let data = abi_encode_params(
+            TRANSFER_SELECTOR,
+            args,
+        );
+
+        let value = zero(get_metadata());
+        let result = evm_call(caller, token, value, data);
+        assert!(is_result_success(&result), error::aborted(ENOT_SUCCESS));
+        emit_evm_logs(&result);
+        result
+    }    
+
+    /// Same as `transfer`, but allowed to be called as an entry function.
+    public entry fun transfer_entry(
+        caller: &signer,
+        token: address,
+        recipient: address,
+        value: u256,
+    ) {
+        transfer(caller, token, recipient, value);
+    }
+
+    struct TransferFromArgs {
+        sender: address,
+        recipient: address,
+        value: u256,
+    }
+
+    public fun transfer_from(
+        caller: &signer,
+        token: address,
+        sender: address,
+        recipient: address,
+        value: u256,
+    ): EvmResult {
+        let args = TransferFromArgs {
+            sender,
+            recipient,
+            value,
+        };
+
+        let data = abi_encode_params(
+            TRANSFER_FROM_SELECTOR,
+            args,
+        );
+
+        let value = zero(get_metadata());
+        let result = evm_call(caller, token, value, data);
+        assert!(is_result_success(&result), error::aborted(ENOT_SUCCESS));
+        emit_evm_logs(&result);
+        result
+    }    
+    
+    struct EmptyArgs {}
```

Comment:
I do not think this is necessary. On the one hand, it is consistent with the pattern laid out for methods with arguments, but on the other hand, bothering to call `abi_encode_params` with an empty struct feels a bit wasteful on Move gas. I think for functions without any arguments it is sufficient to simply pass the selector as the data for `evm_call`.

Reply:
Nice, I didn't think about it from gas perspective, and just followed L2 contract examples. I've created #316 to address that.